encodings/alp/src/alp/compute/compare.rs

```diff
@@ -68,7 +68,7 @@ where
     match encoded {
         Some(encoded) => {
             let s = ConstantArray::new(encoded, alp.len());
-            Ok(Some(compare(alp.encoded(), &s, operator)?))
+            Ok(Some(compare(alp.encoded(), s.as_ref(), operator)?))
```

Comment:
I was curious if we can avoid using as_ref() explicitly and `&*` will work as well, i.e. we reborrow it as the correct type.